packages/shylu/shylu_dd/bddc/example/example1.C

```diff
@@ -0,0 +1,210 @@
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov) 
+// 
+// ************************************************************************
+//@HEADER
+
+#include <mpi.h>
+#include "ShyLU_DDBDDC_config.h"
+#if defined(HAVE_SHYLU_DDBDDC_SHYLU_NODETACHO)
+#include "Kokkos_Core.hpp"
+#endif
+#include "shylu_PreconditionerBDDC.h"
+#include "shylu_KrylovSolver.h"
+#include "setupTest.h"
+#include <Teuchos_XMLParameterListHelpers.hpp>
+
+using Teuchos::RCP;
+
+typedef int LO; // Local Ordinal
+typedef long GO; // Global Ordinal
+typedef double SX; // floating point data type
+typedef double SM; // real (magnitude) for SX
+
+void readInputFiles(MPI_Comm Comm,
+		    const std::string fileNamePM, 
+		    const std::string fileNameBDDC, 
+		    const std::string fileNameMueLu, 
+		    const std::string fileNameNodalAMG, 
+		    RCP<Teuchos::ParameterList> & parametersPM,
+		    RCP<Teuchos::ParameterList> & parametersBDDC,
+		    RCP<Teuchos::ParameterList> & parametersMueLu,
+		    RCP<Teuchos::ParameterList> & parametersNodalAMG)
+{
+  RCP<const Teuchos::Comm<int> > TComm = 
+    Teuchos::rcp( new Teuchos::MpiComm<int>(Comm) );
+  Teuchos::ParameterList paramsPM, paramsBDDC, paramsMueLu, paramsNodalAMG;
+
+  Teuchos::updateParametersFromXmlFileAndBroadcast
+    (fileNamePM, Teuchos::Ptr<Teuchos::ParameterList>(&paramsPM), *TComm);
+  parametersPM = Teuchos::rcp( new Teuchos::ParameterList(paramsPM) );
+  
+  Teuchos::updateParametersFromXmlFileAndBroadcast
+    (fileNameBDDC, Teuchos::Ptr<Teuchos::ParameterList>(&paramsBDDC), *TComm);
+  parametersBDDC = Teuchos::rcp( new Teuchos::ParameterList(paramsBDDC) );
+
+  if (fileNameMueLu != "") {
+    Teuchos::updateParametersFromXmlFileAndBroadcast
+      (fileNameMueLu, Teuchos::Ptr<Teuchos::ParameterList>(&paramsMueLu), 
+       *TComm);
+    parametersMueLu = Teuchos::rcp( new Teuchos::ParameterList(paramsMueLu) );
+  }
+
+  if (fileNameNodalAMG != "") {
+    Teuchos::updateParametersFromXmlFileAndBroadcast
+      (fileNameNodalAMG, Teuchos::Ptr<Teuchos::ParameterList>(&paramsNodalAMG),
+       *TComm);
+    parametersNodalAMG = 
+      Teuchos::rcp( new Teuchos::ParameterList(paramsNodalAMG) );
+  }
+}
+
+void runTest(RCP<Teuchos::ParameterList> & parametersPM,
+	     RCP<Teuchos::ParameterList> & parametersBDDC,
+	     RCP<Teuchos::ParameterList> & parametersMueLu,
+	     RCP<Teuchos::ParameterList> & parametersNodalAMG,
+	     const std::string meshDataFile,
+	     const int myPID,
+	     MPI_Comm Comm,
+	     const bool resetFile,
+	     int & numIterations)
+{
+  // setup test data
+  bddc::setupTest<LO,GO,SX,SM> 
+    test(parametersPM, parametersBDDC, parametersMueLu,
+	 parametersNodalAMG, meshDataFile, Comm);
+  LO numNode, *nodeBegin(nullptr), *localDofs(nullptr);
+  const GO *nodeGlobalIDs(nullptr);
+  const SM *xCoord(nullptr), *yCoord(nullptr), *zCoord(nullptr);
+  std::vector<LO*> subRowBeginPtr, subColumnsPtr;
```

Comment:
`std::vector` of raw pointer suggests a memory leak to me.  Do those inner pointers get freed / cleaned up at some point?